testdata/deployment.json

```diff
@@ -35,6 +35,7 @@
                 "condition": true
             },
             "prop_string": "hello world",
+            "repo": "github.com:hashicorp/vault",
```

Comment:
Ensure duplicate repos are not added.

Reply:
Wouldn't this be invalid JSON since `"repo"` is a duplicate key?

Reply:
@davidbloss thanks for pointing that out - it isn't valid JSON and also that makes the tests ineffective at checking for duplicates. 

I added a commit with 2 files: duplicate_expectation.json and duplicate_config.json. The config adds every jq expression result twice. 

I validated it work by purposefully breaking the test,

```
// set.go modified
// will not deduplicate correctly because unique key keeps changing
func randomString(length int) string {
	rand.Seed(time.Now().UnixNano())
	b := make([]byte, length+2)
	rand.Read(b)
	return fmt.Sprintf("%x", b)[2 : length+2]
}

// Add performs deduplication of objects in the Set by treating the json marshaled value as a unique key
func (set Set[T]) Add(object T) {
	marshaled, err := json.Marshal(&object)
	if err != nil {
		return
	}
	set.data[string(marshaled)+randomString(2)] = object
}





=== RUN   TestJQServiceParser/using_simple_config
=== RUN   TestJQServiceParser/using_sample_config
=== RUN   TestJQServiceParser/using_duplicate_config
    jq_service_parser_test.go:77: 
                exp: opslevel_jq_parser.ServiceRegistration{Aliases:[]string{"k8s:web-self-hosted", "self-hosted-web"}, Description:"", Framework:"", Language:"", Lifecycle:"", Name:"web", Owner:"", Product:"", Properties:map[string]string{}, Repositories:[]opslevel.ServiceRepositoryCreateInput{opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00029e320)}, BaseDirectory:(*string)(0xc00029e340), DisplayName:(*string)(0xc00029e350)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00029e360)}, BaseDirectory:(*string)(0xc00029e370), DisplayName:(*string)(0xc00029e390)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00029e3b0)}, BaseDirectory:(*string)(0xc00029e3c0), DisplayName:(*string)(0xc00029e3e0)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00029e3f0)}, BaseDirectory:(*string)(0xc00029e420), DisplayName:(*string)(0xc00029e430)}}, System:"", TagAssigns:[]opslevel.TagInput{opslevel.TagInput{Key:"app.kubernetes.io/instance", Value:"web"}, opslevel.TagInput{Key:"app.kubernetes.io/part-of", Value:"opslevel"}, opslevel.TagInput{Key:"has_rbac", Value:"yes"}, opslevel.TagInput{Key:"has_runtime", Value:"no"}, opslevel.TagInput{Key:"imported", Value:"kubectl-opslevel"}, opslevel.TagInput{Key:"is_fork", Value:"no"}, opslevel.TagInput{Key:"kots.io/app-slug", Value:"opslevel"}, opslevel.TagInput{Key:"kots.io/backup", Value:"velero"}, opslevel.TagInput{Key:"monolithic", Value:"yes"}}, TagCreates:[]opslevel.TagInput{opslevel.TagInput{Key:"environment", Value:"dev"}}, Tier:"", Tools:[]opslevel.ToolCreateInput{opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-ci", Url:"https://circleci.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-graphs", Url:"https://datadog.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-logs", Url:"https://splunk.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-schedule", Url:"https://pagerduty.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}}}
                got: opslevel_jq_parser.ServiceRegistration{Aliases:[]string{"k8s:web-self-hosted", "k8s:web-self-hosted", "self-hosted-web", "self-hosted-web"}, Description:"", Framework:"", Language:"", Lifecycle:"", Name:"web", Owner:"", Product:"", Properties:map[string]string{}, Repositories:[]opslevel.ServiceRepositoryCreateInput{opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00009de40)}, BaseDirectory:(*string)(0xc00009de60), DisplayName:(*string)(0xc00009de70)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00009de80)}, BaseDirectory:(*string)(0xc00009dea0), DisplayName:(*string)(0xc00009deb0)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00009dec0)}, BaseDirectory:(*string)(0xc00009dee0), DisplayName:(*string)(0xc00009def0)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc000026010)}, BaseDirectory:(*string)(0xc000026030), DisplayName:(*string)(0xc000026060)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00009df00)}, BaseDirectory:(*string)(0xc00009df20), DisplayName:(*string)(0xc00009df30)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc000026080)}, BaseDirectory:(*string)(0xc000026120), DisplayName:(*string)(0xc000026170)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00009de00)}, BaseDirectory:(*string)(0xc00009de20), DisplayName:(*string)(0xc00009de30)}, opslevel.ServiceRepositoryCreateInput{Service:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(nil)}, Repository:opslevel.IdentifierInput{Id:(*opslevel.ID)(nil), Alias:(*string)(0xc00009df40)}, BaseDirectory:(*string)(0xc00009df60), DisplayName:(*string)(0xc00009df70)}}, System:"", TagAssigns:[]opslevel.TagInput{opslevel.TagInput{Key:"app.kubernetes.io/instance", Value:"web"}, opslevel.TagInput{Key:"app.kubernetes.io/instance", Value:"web"}, opslevel.TagInput{Key:"app.kubernetes.io/part-of", Value:"opslevel"}, opslevel.TagInput{Key:"app.kubernetes.io/part-of", Value:"opslevel"}, opslevel.TagInput{Key:"has_rbac", Value:"yes"}, opslevel.TagInput{Key:"has_rbac", Value:"yes"}, opslevel.TagInput{Key:"has_runtime", Value:"no"}, opslevel.TagInput{Key:"has_runtime", Value:"no"}, opslevel.TagInput{Key:"imported", Value:"kubectl-opslevel"}, opslevel.TagInput{Key:"imported", Value:"kubectl-opslevel"}, opslevel.TagInput{Key:"is_fork", Value:"no"}, opslevel.TagInput{Key:"is_fork", Value:"no"}, opslevel.TagInput{Key:"kots.io/app-slug", Value:"opslevel"}, opslevel.TagInput{Key:"kots.io/app-slug", Value:"opslevel"}, opslevel.TagInput{Key:"kots.io/backup", Value:"velero"}, opslevel.TagInput{Key:"kots.io/backup", Value:"velero"}, opslevel.TagInput{Key:"monolithic", Value:"yes"}, opslevel.TagInput{Key:"monolithic", Value:"yes"}}, TagCreates:[]opslevel.TagInput{opslevel.TagInput{Key:"environment", Value:"dev"}, opslevel.TagInput{Key:"environment", Value:"dev"}}, Tier:"", Tools:[]opslevel.ToolCreateInput{opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-ci", Url:"https://circleci.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-ci", Url:"https://circleci.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-graphs", Url:"https://datadog.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-graphs", Url:"https://datadog.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-logs", Url:"https://splunk.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-logs", Url:"https://splunk.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-schedule", Url:"https://pagerduty.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}, opslevel.ToolCreateInput{Category:"logs", DisplayName:"my-schedule", Url:"https://pagerduty.com", Environment:(*string)(nil), ServiceId:(*opslevel.ID)(nil), ServiceAlias:(*string)(nil)}}}
--- FAIL: TestJQServiceParser (0.03s)
    --- PASS: TestJQServiceParser/using_simple_config (0.01s)
    --- PASS: TestJQServiceParser/using_sample_config (0.01s)
    --- FAIL: TestJQServiceParser/using_duplicate_config (0.01s)

```